File: src/main/java/it/unibo/crabinv/Model/powerUpsShop/PowerUpFactory.java
package it.unibo.crabinv.Model.powerUpsShop;

import java.util.List;

public final class PowerUpFactory {

    private PowerUpFactory() {}

    public static List<PowerUp> createShopPowerUps() {
        return List.of(
                new PowerUpLogic(PowerUpType.SPEED_UP, 150, 3),
                new PowerUpLogic(PowerUpType.FIRERATE_UP, 100, 5),
                new PowerUpLogic(PowerUpType.HEALTH_UP, 200, 4)
        );
    }
}

File: src/main/java/it/unibo/crabinv/model/powerups/PowerUpFactory.kt
package it.unibo.crabinv.model.powerups

/**
 * It's the factory of PowerUps.
 */
object PowerUpFactory {
    private const val SPEED_COST = 150
    private const val SPEED_MAXLEVEL = 3
    private const val FIRERATE_COST = 100
    private const val FIRERATE_MAXLEVEL = 5
    private const val HEALTH_COST = 200
    private const val HEALTH_MAXLEVEL = 4

    /**
     * Creates the power Ups and makes them a list.
     * @return the list of avaible power ups
     */
    fun createShopPowerUps(): List<PowerUp> =
        listOf<PowerUp>(
            PowerUpLogic(PowerUpType.SPEED_UP, SPEED_COST, SPEED_MAXLEVEL),
            PowerUpLogic(PowerUpType.FIRERATE_UP, FIRERATE_COST, FIRERATE_MAXLEVEL),
            PowerUpLogic(PowerUpType.HEALTH_UP, HEALTH_COST, HEALTH_MAXLEVEL),
        )
}
